src/java_tools/junitrunner/java/com/google/testing/coverage/JacocoCoverageRunner.java

```diff
@@ -345,7 +346,30 @@ private static String getUniquePath(String pathTemplate, String suffix) throws I
     return convertedMetadataFiles.build();
   }
 
-  private static URL[] getUrls(ClassLoader classLoader) {
+  private static URL[] getUrls(ClassLoader classLoader, boolean wasWrappedJar) {
+    URL[] urls = getClassLoaderUrls(classLoader);
+    // If the classpath was too long then a temporary top-level jar is created containing nothing but a manifest with
+    // the original classpath. For the moment, we just assume it's made up of URLs to other jars and extract them.
+    if (urls != null) {
+      if (wasWrappedJar && urls.length == 1) {
```

Comment:
nit: Can you merge those two together: `if (wasWrappedJar && urls != null && urls.length ==1)`. This will reduce for one block ident and have urls parameters together, which is easier to read.